porting oh-my-posh to Illumos and Solaris

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What would you like to see changed?

I'm interested in trying to port oh-my-posh to Illumos (opensource Solaris, I'm using the OpenIndiana distro but one port should be fine for the other distros and likely commercial Solaris too).  I've skimmed the docs and I see good documentation on adding a new feature (segment), but I don't think that covers adding a new OS/platform.  I'm wondering if you can provide some guidance on how various implementations are selected?  I see the build comments at the top of some Go source files, but I'm still a little confused on how a particular implementation is being selected.

Illumos has recent go-lang and bash and zsh.  I've tried both powerline and a bash implementation of oh-my-zsh, and they both work on Illumos, but they're slow.  Hence the interest in oh-my-posh.

Right now `go build -v` fails with

```
$ go build -v
github.com/jandedobbeleer/oh-my-posh/src/runtime/battery
# github.com/jandedobbeleer/oh-my-posh/src/runtime/battery
runtime/battery/battery_windows_nix.go:60:20: undefined: systemGetAll
```

Using Illumos on a laptop isn't all that common, so to start with I would like to just add a no-op for the battery status.  I might be able to improve that later if I can figure out what MATE's battery indicator app is calling in C.

I'm not super proficient in Go, but I have done some small projects and the benefit here is that I have working implementations for various BSDs and Linux to look at for examples.  I will definitely need some guidance, though.

--- a/src/runtime/battery/battery_illumos.go
+++ b/src/runtime/battery/battery_illumos.go
@@ -1,0 +1,11 @@
+//go:build illumos || solaris
+
+package battery
+
+// Get returns information about all batteries in the system.
+//
+// Battery reporting is not implemented on illumos/solaris; this always
+// returns a NoBatteryError so the battery segment disables itself cleanly.
+func Get() (*Info, error) {
+	return nil, &NoBatteryError{}
+}

--- a/src/runtime/battery/battery_illumos_test.go
+++ b/src/runtime/battery/battery_illumos_test.go
@@ -1,0 +1,16 @@
+//go:build illumos || solaris
+
+package battery
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetNoBattery(t *testing.T) {
+	info, err := Get()
+
+	assert.Nil(t, info)
+	assert.IsType(t, &NoBatteryError{}, err)
+}

--- a/src/runtime/battery/battery_windows_nix.go
+++ b/src/runtime/battery/battery_windows_nix.go
@@ -1,4 +1,4 @@
-//go:build !darwin && !netbsd && !openbsd && !freebsd && !js
+//go:build !darwin && !netbsd && !openbsd && !freebsd && !js && !illumos && !solaris
 
 package battery
 

--- a/src/runtime/battery/battery_windows_nix_test.go
+++ b/src/runtime/battery/battery_windows_nix_test.go
@@ -1,4 +1,4 @@
-//go:build !darwin && !netbsd && !openbsd
+//go:build !darwin && !netbsd && !openbsd && !illumos && !solaris
 
 // battery
 // Copyright (C) 2016-2017 Karol 'Kenji Takahashi' Woźniak

